test/show.jl

```diff
@@ -730,8 +730,8 @@ let x = [], y = [], z = Base.ImmutableDict(x => y)
     push!(x, y)
     push!(y, x)
     push!(y, z)
-    @test replstr(x) == "1-element Array{Any,1}:\n Any[Any[Any[#= circular reference @-2 =#]], Base.ImmutableDict(Any[Any[#= circular reference @-3 =#]]=>Any[#= circular reference @-2 =#])]"
-    @test repr(z) == "Base.ImmutableDict(Any[Any[Any[#= circular reference @-2 =#], Base.ImmutableDict(#= circular reference @-3 =#)]]=>Any[Any[Any[#= circular reference @-2 =#]], Base.ImmutableDict(#= circular reference @-2 =#)])"
```

Comment:
This change is not really wanted, but is necessary to make the tests pass. I think it's due to the fact that `Any` is generally the default value for `typeinfo`, but we may instead nead to use `nothing` (or `missing`?). I will investigate and open a separate PR if needed.

Reply:
So this is due to the fact that we show `Dict` easily without type information (when the types are concrete), e.g. compare on master:
```julia
julia> show(Int8[1])
Int8[1]

julia> show(Dict(Int8(1)=>1))
Dict(1=>1)
```
I.e. compared to arrays, we loose information when showing a `Dict`. I don't find this ideal, but this is not directly related to this PR.